dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/FdcService.java

```diff
@@ -168,123 +181,118 @@ private Integer updateFdcAndCreateFile(List<Fdc> successfulFdcs, Map<String, Str
             if (!failedFdcs.isEmpty()) {
                 log.info("Failed to send {} FDC contributions", failedFdcs.size());
             }
-            // Setup and make MAAT API "ATOMIC UPDATE" REST call below:
-            try {
-                contributionFileId = fdcUpdateRequest(xmlFile, successfulIdList, successfulIdList.size(), fileName, ackXml);
-                // Explicitly log the FDC contribution IDs that were updated:
-                log.info("Created FDC contribution-file ID {} from {} FDC contribution IDs [{}]", contributionFileId, successfulIdList.size(), String.join(", ", successfulIdList));
-            } catch (MaatApiClientException e){
-                handleFileCreationError(e, e.getStatusCode());
-                throw e;
-            } catch (WebClientResponseException e){
-                handleFileCreationError(e, e.getStatusCode());
-                throw e;
-            } catch (HttpServerErrorException e){
-                handleFileCreationError(e, e.getStatusCode());
-                throw e;
-            }
+            contributionFileId = executeFdcUpdateRequestCall(xmlFile, successfulIdList, successfulIdList.size(), fileName, ackXml);
         }
-        logMaatUpdateEvents(successfulFdcs, failedFdcs);
+        logMaatUpdateEvent(successfulFdcs, failedFdcs);
         return contributionFileId;
     }
 
-    private void handleFileCreationError(Exception e, HttpStatusCode httpStatusCode) {
-        // We're rethrowing the exception, therefore avoid logging the stack trace to prevent logging the same trace multiple times.
-        log.error("Failed to create FDC contribution-file. Investigation needed. State of files will be out of sync! [" + e.getClass().getName() + "(" + e.getMessage() + ")]");
-        // If failed, we want to handle this. As it will mean the whole process failed for current day.
-        eventService.logFdc(UPDATED_IN_MAAT, batchId,null, httpStatusCode, e.getMessage());
-    }
+    // External Call Executions Methods
 
-    public void logNumberDiscepancies(int globalUpdateCount, int getFdcCount, int successfullySentFdcCount) {
-        if (globalUpdateCount != getFdcCount) {
-            log.info("FDC contribution count discrepancy: {} affected by FDC global update, {} from getFdcs", globalUpdateCount, getFdcCount);
-        }
-        if (getFdcCount != successfullySentFdcCount) {
-            log.info("FDC contribution count discrepancy: {} from getFdcs, {} successfully sent", getFdcCount, successfullySentFdcCount);
+    private int executeFdcUpdateCall(UpdateLogFdcRequest updateLogFdcRequest) {
```

Comment:
As this is all about refactoring, can we rename some of the very similarly named methods, viz. 

executeFdcUpdateCall (which seems to be about logging), executeFdcUpdateRequestCall (which I think creates the FDC file), executeFdcGlobalUpdateCall (which can stay)

Reply:
Good call. Will also rename the processFdcUpdate() method as well. It's not obvious from the name it's handling a DRC Processed Request.

Reply:
The first execute listed was actually to encapsulate the feature stubbing, so that it'd be nice and separate from the process at large.